src/solver/optimisation/opt_alloc_probleme_a_optimiser.cpp

```diff
@@ -74,7 +74,7 @@ void OPT_AllocateFromNumberOfVariableConstraints(PROBLEME_ANTARES_A_RESOUDRE* Pr
     ProblemeAResoudre->TypeDeVariable.assign(szNbVarsint, 0);
     ProblemeAResoudre->Xmin.assign(szNbVarsDouble, 0.);
     ProblemeAResoudre->Xmax.assign(szNbVarsDouble, 0.);
-    ProblemeAResoudre->X = (double*)MemAlloc(szNbVarsDouble);
+    ProblemeAResoudre->X.assign(szNbVarsDouble, 0.);
```

Comment:
```suggestion
    ProblemeAResoudre->X.assign(nbVariables);
```
No need to multiply by sizeof `type`